examples/buttons_wait_for.rs

```diff
@@ -0,0 +1,44 @@
+//! An extremely simple libtock-rs example. Register button events.
+#![no_main]
+#![no_std]
+
+use core::fmt::Write;
+use libtock::buttons::{ButtonListener, Buttons};
+use libtock::console::Console;
+use libtock::leds::Leds;
+use libtock::runtime::{set_main, stack_size};
+use libtock_platform::share;
+// use libtock_runtime::TockSyscalls;
+
+set_main! {main}
+stack_size! {0x1000}
+
+fn main() {
+    writeln!(Console::writer(), "main!").unwrap();
+    let listener = ButtonListener(|button, _state| {
+        let _ = Leds::toggle(button);
+        // writeln!(Console::writer(), "button {:?}: {:?}", button, state).unwrap();
+    });
+    if let Ok(buttons_count) = Buttons::count() {
+        writeln!(Console::writer(), "button count: {}", buttons_count).unwrap();
+
+        share::scope(|subscribe| {
+            // Subscribe to the button callback.
+            Buttons::register_listener(&listener, subscribe).unwrap();
```

Comment:
`yield-wait-for` does not trigger the callback, it returns the arguments for the callback when `yield-for-returns`.